--- efl-webkit/Source/JavaScriptCore/profiler/ProfileNode.cpp~	2014-02-02 04:21:06.000000000 +0000
+++ efl-webkit/Source/JavaScriptCore/profiler/ProfileNode.cpp	2016-05-28 08:39:06.597242990 +0000
@@ -123,7 +123,7 @@
 {
     ASSERT(!m_calls.isEmpty());
 
-    if (isnan(m_calls.last().totalTime()))
+    if (std::isnan(m_calls.last().totalTime()))
         endAndRecordCall();
 
     // Because we iterate in post order all of our children have been stopped before us.
@@ -147,7 +147,7 @@
 void ProfileNode::endAndRecordCall()
 {
     Call& last = lastCall();
-    ASSERT(isnan(last.totalTime()));
+    ASSERT(std::isnan(last.totalTime()));
 
     last.setTotalTime(currentTime() - last.startTime());
 
